spec/lib/scheduler/thread_pool_spec.rb

```diff
@@ -0,0 +1,164 @@
+# frozen_string_literal: true
+
+RSpec.describe Scheduler::ThreadPool do
+  let(:min_threads) { 2 }
+  let(:max_threads) { 4 }
+  let(:idle_time) { 0.1 }
+
+  let(:pool) do
+    described_class.new(min_threads: min_threads, max_threads: max_threads, idle_time: idle_time)
+  end
+
+  after do
+    begin
+      pool.shutdown
+    rescue StandardError
+      nil
```

Comment:
When an error is encountered, doesn't it mean we are not cleaning up properly? Would prefer if we can rescue all errors here as it is an anti-pattern.

Reply:
true... this is bad... in my defense this was one of the things I missed cleaning up some claude code :)